### DISCUSSION TITLE
Grabbing connection string from Azure Config

### DISCUSSION THREAD

**Discussion by mrpavlitzky** [OrchardCMS/OrchardCore#11955] [Q&A] [answered]
Hi! 

I'm trying to secure the db connection string and inject it from the Azure application settings in production. I followed the steps prescribed by @deanmarcussen on issue #3452, but it doesn't seem to work for me.

So what I did is put an incorrect connection string (just for tests) in my App_Data/Sites/Default/appsettings.json file.
Then I added a correct one as an Azure app setting: 
Orchard__Default__ConnectionString = "correctOne"
Orchard__Default__State = " "

In this configuration the app is not working. I checked with Kudu and the environment variables are correctly recognized, while from the eventlog dump I can see that the application still wants to connect using the incorrect connection string defined in APP_DATA. 

Is there any other configuration that needs to be performed to have it recognize the environment variable defined on Azure side?

Cheers

**Comment by hishamco** [comment]
If I'm not wrong the connection string are protected using secrets, I saw a demo long time back by @deanmarcussen 

Did you checked https://github.com/OrchardCMS/OrchardCore/blob/339d74b251b53393dd510f8ad3027aa17d78500d/src/docs/reference/core/Shells/README.md

**Reply by mrpavlitzky** [reply]
Thank you for your reply. This is not quite what I'm looking for. The documentation you provided states:

`The primary purpose of the Shell Configuration providers is to provide a shared external environment for multi-tenancy where tenants need to be created, and their settings mutated, during live operation of the stateless hosts.`

On issue #3452 @deanmarcussen gives instructions on configuring Azure app settings without any reference to the blob storage. I followed the steps closely, but cannot quite get it working.

**Reply by mrpavlitzky** [reply]
Anyway, from what I'm seeing in the Azure Shells Configuration you are to provide a connection string in the appsettings.json to the Azure Storage Account which by itself should be protected and ideally inserted at runtime from a safe Azure configuration space. So due to safety reasons the question of configuring Azure app settings still persits.

**Comment by mrpavlitzky** [comment, marked as answer]
Solved:

The individual tenant settings override the environment variables. Deleting the connection string from App_Data/Sites/Default/appsettings.json solves the issue. I got confused by the documentation, see: #11956
